src/run.c

```diff
@@ -109,7 +109,6 @@ static void run(uint32_t outlen, char *pwd, char *salt, uint32_t t_cost,
         fatal("salt missing");
     }
 
-    pwdlen = strlen(pwd);
```

Comment:
You are right. I wasn't catching it since when `strlen(pwd)` = 128 it didn't run the function, so there was always a `\0` at the end. Nevertheless I changed since the function call was unnecessary.